TFP-6342 Endring opphold svangerskapspenger

AnjaAalerud opened the pull request:
- Oppholdsperioden avkortes ikke selv om ny svangerskapspengeperiode starter midt i oppholdet. Det kan bli feil for opphold pga 100% sykepenger. 
- Opphold som beholdes fra forrige søknad får kilde Tidligere vedtak på samme måte som svangerskapsperiodene. 
- Henter opprinnelig oppholdkilde dersom den er TIDLIGERE_VEDTAK. Settes til saksbehandler om ingen match
- Forhindrer duplikater dersom forrige og nytt opphold er før ny tilretteleggingFom dato

AnjaAalerud committed:
TFP-6342 Endring opphold svangerskapspenger

*Oppholdsperioden avkortes ikke selv om ny svangerskapspengeperiode starter midt i oppholdet. Det kan bli feil for opphold pga 100% sykepenger.
*Opphold som beholdes fra forrige søknad får kilde Tidlgere vedtak på samme måte som svangerskapsperiodene.
*Fjernet at oppholdkilde mappes til fpoversikt da denne ikke ble brukt.

palfi reviewed (commented):
Tror du må fjerne fra fpoversikt først, ellers blir det nullpointer

jolarsen commented on domenetjenester/mottak/src/main/java/no/nav/foreldrepenger/mottak/dokumentpersiterer/impl/søknad/v3/TilretteleggingOversetter.java:
```diff
@@ -148,24 +149,24 @@ private SvpTilretteleggingEntitet oppdaterEksisterendeTlrMedNyeFomsOgOpphold(Svp
+            .filter(eksOpphold -> eksOpphold.getFom().isBefore(tidligsteNyFom))
+            .map(eksOpphold ->
+                    SvpAvklartOpphold.Builder.nytt()
+                        .medKilde(SvpOppholdKilde.TIDLIGERE_VEDTAK)
```
Hm, vil du da ikke kunne få få 2 sett med like perioder - en med tidligere vedtak og en med søknad?
Eksempel: Sak fra TFP, ferien er lik i de to søknadene, men slår litt ut med tidligsteFom.
Ganske usikker på dette området :-)

jolarsen reviewed (approved):
Har du nok tester som har varianter av opphold (like i alle søknadene/vedtak) der ferie
* Begynner etter opprinneligTidligsteFom men før nyTidligsteFom
* Begynner etter opprinneligTidligsteFom og etter nyTidligsteFom

AnjaAalerud committed:
TFP-6342 Endring opphold svangerskapspenger

Oppholdsperioden avkortes ikke selv om ny svangerskapspengeperiode starter midt i oppholdet. Det kan bli feil for opphold pga 100% sykepenger.
Opphold som beholdes fra forrige søknad får kilde Tidlgere vedtak på samme måte som svangerskapsperiodene.
Lagt til at vi henter opprinnelig oppholdkilde fra forrige behandling om finnes. Hvis den ikke finnes settes SAKSBEHANDLER som kilde.

AnjaAalerud commented on web/src/main/java/no/nav/foreldrepenger/web/app/tjenester/fpoversikt/SvpDtoTjeneste.java:
```diff
@@ -213,12 +217,61 @@ private Set<SvpSak.OppholdPeriode> finnAlleOppholdsperioderFraTlr(SvpTilretteleg
+                .orElse(Collections.emptyList())
+                .stream()
+                .flatMap(forrige -> forrige.getAvklarteOpphold().stream())
+                .filter(forrigeOpphold -> forrigeOpphold.getFom().equals(gjeldendeOpphold.getFom()) && forrigeOpphold.getTom()
```
Hvis de ikke er like er det fordi saksbehandler har endret på et eller annet tidspunkt etter de ble kopiert. Hvis vi ikke finner den settes kilde derfor til Saksbehandler.

AnjaAalerud committed:
TFP-6342 Endring opphold svangerskapspenger

Fjernet TIDLIGERE_VEDTAK fra SvpSak.OppholdPeriode da den aldri skal vises til bruker i deres oversikt.

AnjaAalerud committed:
TFP-6342 Endring opphold svangerskapspenger

Beholder IKKE eksisterende opphold dersom nytt opphold har samme datoer og oppholdårsak som eksisterende. Lagt inn tester for dette

AnjaAalerud committed:
Merge branch 'master' into TFP-6342
